src/ui/win32/DialogBase.cpp

```diff
@@ -20,47 +22,58 @@ DialogBase::~DialogBase() noexcept
     }
 }
 
-static INT_PTR CALLBACK StaticDialogProc(HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
+_Use_decl_annotations_
+_NODISCARD static INT_PTR CALLBACK StaticDialogProc(HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
 {
-    DialogBase* pDialog = reinterpret_cast<DialogBase*>(GetWindowLongPtr(hDlg, DWLP_USER));
+    const auto pDialog{ reinterpret_cast<DialogBase*>(GetWindowLongPtr(hDlg, DWLP_USER)) };
+
+    // TBD: A dialog should not be calling DefWindowProc, it should return 0 
     if (pDialog == nullptr)
-        return DefWindowProc(hDlg, uMsg, wParam, lParam);
+        return ::DefWindowProc(hDlg, uMsg, wParam, lParam);
 
-    INT_PTR result = pDialog->DialogProc(hDlg, uMsg, wParam, lParam);
+    const auto result{ pDialog->DialogProc(hDlg, uMsg, wParam, lParam) };
```

Comment:
Why do you insist on reformatting everything? There's no benefit to this change, and it just makes it harder for me to review.

Reply:
Well, now this function is inconsistent with itself. Line 28 still has braces. I'm more concerned that you're changing code just because you don't like the way it's formatted. 

I don't like brace initialization for anything that's not a container. "name=value" is the standard way to initialize a variable in almost every language. Just because c++11 introduced another way to do things doesn't mean we should. I have roughly 20 years of professional programming experience (about half of that in C/C++), and I struggle with some of the stuff you're trying to push through. I can only imagine how off-putting this would be to someone who barely understands how templates work. I want to keep the code approachable. Chances are I won't be working on the project in a few years, and I don't want it to die because the code is too complex to read for an "average" developer.

Reply:
I did have a branch "ready" (like yesterday about const correctness) but because of all your feedback I didn't feel like it was actually ready because all my brace initialization. I do believe there is a benefit to brace initialization as it skips the copy/move assignment step but it's not that much of a benefit for primitives and pointers. I probably won't work on this in a few years either.

I totally understand your point of view. I can see you're point about simple stuff like braces, initialization, comments, and formatting. You definitely have more in the experience department and probably in the life department as well. I'm "only" 26, and I still have a lot to learn. I do take your advice to heart.

Templates though, the entire C++ Standard Library is full of them. Template meta-programming and abstractions was one of the very first things I learned when doing C++ for ADT structures (stacks, queues, trees, etc.). At the time C++11 (around 2014) was new and that's how everyone I knew (in-person), amateurs and professionals alike including my dad whose in his 60s.

I've always been told to get on the cutting edge of technology to stay coherent as a software engineer. The formatting thing though, I understand that.

Reply:
About the formatting thing. Visual Studio 2017 has the option of using ClangFormat (`tools->options->Text Editor->C/C++->Formatting`).

It's kind of like an advanced editorconfig. There are default standards, but from all previous disscusions I really doubt any of them will satisfy the CONTRIBUTING file. The default standard are `LLVM, Google, Chromium, Mozilla and WebKit`. I tried them all before but they are closer to what I use, which seems to be considered unacceptable. There is an option to use a custom clang-format.exe file, to automate the custom formatting standard specified. That could be a solution.